kubernetes/cmsweb/docs/dmwm-documentation.md

```diff
@@ -0,0 +1,206 @@
+      RewriteRule ^(/ms-pileup(/.*)?)$ https://%{SERVER_NAME}${escape:$1}%{env:CMS_QUERY} [R=301,NE,L]
+    ```
+    AND
+    ```
+        [_frontend@cmsweb-testbed-v1-22-zone-a-whjk2m547jvl-node-0 data]$ cat srv/current/config/frontend/backends.txt | grep ms-pileup
+    ^/auth/complete/ms-pileup(?:/|$) ms-pileup.dmwm.svc.cluster.local
+    ```
+  - If they don't exist in backends.txt, update the secrets by applying the services_config files mentioned above, and restart the frontend daemonset.
+  - If they don't exist in the server.conf or `srv/current/config/frontend/app\_\<service-name\>\_ssl/nossl.conf` files, and you are sure your changes were incorporate, then maybe your cluster is running an older version of the frontend service.
+
+## Health/Load of the cluster
+
+1. **Get Cluster Information**:
+   - To get general information about your cluster, including its version and nodes, use: `kubectl cluster-info`.
+
+2. **Check Node Status**:
+   - To view the status of your cluster nodes and their resource utilization, you can use: `kubectl get nodes`.
+
+3. **View Pod Status**:
+   - To check the status of pods running in your cluster: `kubectl get pods -n <namespace>`.
+
+4. **Get Resource Usage**:
+   - Use `kubectl top` to check resource usage for nodes or pods. For example, to view CPU and memory usage for pods: `kubectl top pods -n <namespace>`.
+   - To view node resource usage: `kubectl top nodes`.
+
+5. **Check Cluster Events**:
+   - To see recent cluster events, run: `kubectl get events`.
+
+6. **Examine Logs**:
+   - View logs of a pod for troubleshooting or performance analysis: `kubectl logs <pod-name> -n <namespace>`.
+
+7. **List Deployments**:
+   - List all deployments in a namespace: `kubectl get deployments -n <namespace>`.
+
+8. **Scale Deployments**:
+   - Scale a deployment up or down using `kubectl scale`. For example, to scale a deployment to 3 replicas: `kubectl scale deployment <deployment-name> --replicas=3 -n <namespace>`.
+
+9. **Resource Requests and Limits**:
+   - Check the resource requests and limits of pods with: `kubectl describe pod <pod-name> -n <namespace>`. This helps you understand resource allocation.
+
+10. **Pod Restart Count**:
+    - To see how many times a pod has been restarted, use: `kubectl get pods <pod-name> -n <namespace> -o=jsonpath='{.status.containerStatuses[0].restartCount}'`.
+
+11. **Rolling Restart**:
+    - Perform a rolling restart of a deployment: `kubectl rollout restart deployment/<deployment-name> -n <namespace>`.
+
+12. **Attach to Running Pod**:
+    - Attach to a running pod for interactive troubleshooting: `kubectl exec -it <pod-name> -n <namespace> -- /bin/sh`.
+
+13. **Horizontal Pod Autoscaling**:
+    - To check the status of Horizontal Pod Autoscalers (HPA), use: `kubectl get hpa -n <namespace>`.
+
+14. **Check API Resources**:
+    - Get a list of available API resources with: `kubectl api-resources`.
+
+15. **Check Component Status**:
```

Comment:
Apologies for my ignorance. But what are these components supposed to be? Are they documented somewhere in cmsweb-docs?